paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/refund/PaytrailRefundPaymentService.java

```diff
@@ -245,47 +244,44 @@ private RefundPayment createRefundPayment(
         String now = DateTimeUtil.getDateTime();
 
-        PaymentMessage.PaymentMessageBuilder paymentMessageBuilder = PaymentMessage.builder()
-                .eventType(EventType.PAYMENT_PAID)
-                .eventTimestamp(now)
-                .namespace(payment.getNamespace())
-                .paymentId(payment.getRefundPaymentId())
-                .orderId(payment.getOrderId())
-                .userId(payment.getUserId())
-                .paymentPaidTimestamp(now)
-                .orderType(payment.getRefundType());
+        RefundMessage.RefundMessageBuilder refundMessageBuilder = RefundMessage.builder()
+                .eventType(EventType.REFUND_PAID)
+                .timestamp(now)
+                .namespace(refundPayment.getNamespace())
+                .refundId((refundPayment.getRefundId()))
+                .refundId(refundPayment.getRefundPaymentId())
+                .orderId(refundPayment.getOrderId())
+                .userId(refundPayment.getUserId());
 
-        PaymentMessage paymentMessage = paymentMessageBuilder.build();
+        RefundMessage refundMessage = refundMessageBuilder.build();
 
-        orderPaidWebHookAction(paymentMessage);
-
-        sendEventService.sendEventMessage(TopicName.PAYMENTS, paymentMessage);
-        log.debug("triggered event REFUND_PAID for refundId: " + payment.getRefundPaymentId());
+        refundPaidWebHookAction(refundMessage);
+        log.debug("triggered event REFUND_PAID for refundId: " + refundPayment.getRefundPaymentId());
     }
 
-    protected void orderPaidWebHookAction(PaymentMessage message) {
+    protected void refundPaidWebHookAction(RefundMessage message) {
         try {
-            sendNotificationService.sendPaymentMessageNotification(message);
+            sendNotificationService.sendRefundMessageNotification(message);
         } catch (Exception e) {
             log.error("webhookAction: failed action after receiving event, eventType: " + message.getEventType(), e);
         }
     }
 
-    public void triggerRefundPaymentFailedEvent(Payment payment) {
+    public void triggerRefundPaymentFailedEvent(RefundPayment refund) {
         PaymentMessage paymentMessage = PaymentMessage.builder()
                 .eventType(EventType.PAYMENT_FAILED)
-                .namespace(payment.getNamespace())
-                .paymentId(payment.getPaymentId())
-                .orderId(payment.getOrderId())
-                .userId(payment.getUserId())
+                .namespace(refund.getNamespace())
+                .paymentId(refund.getRefundId())
+                .orderId(refund.getOrderId())
+                .userId(refund.getUserId())
                 //.paymentPaidTimestamp(payment.getTimestamp())
-                .orderType(payment.getPaymentType())
+                .orderType(refund.getRefundType())
                 .build();
         sendEventService.sendEventMessage(TopicName.PAYMENTS, paymentMessage);
```

Comment:
Instead of sending it to kafkas payment topic, send it to ActiveMQ refund-notifications queue by using the same method as above: refundPaidWebHookAction(...). You could rename the method to refundWebHookAction(...) to make it more general for all RefundMessages.